packages/block-library/src/post-content/edit.js

```diff
@@ -2,17 +2,28 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import PostContentInnerBlocks from './inner-blocks';
 
 export default function PostContentEdit( { context: { postId, postType } } ) {
-	if ( postId && postType ) {
+	const { id, type } = useSelect( ( select ) => {
+		return select( 'core/editor' ).getCurrentPost() ?? {};
+	} );
+
+	// Only render InnerBlocks if the context is different from the active post
+	// to avoid infinite recursion of post content.
+	if ( postId && postType && postId !== id && postType !== type ) {
```

Comment:
I'm wondering if it would make sense to move this kind of check into the logic that provdes block context, in order to avoid any other blocks end up with the same infinite recursion issue. The context logic could e.g. set `context` to `{}` in those cases. The individual blocks would then still have to add UI for that case (placeholders, most likely), but they'd be prevented from causing infinite loops.

Reply:
(Not a blocker, we can go ahead with this approach for now.)

However, let's maybe rename the var names in this component, since they're quite confusing right now (`postId` :left_right_arrow: `id`, `postType` :left_right_arrow: `type`)

Reply:
> The context logic could e.g. set context to {} in those cases

The challenging thing here is that the context is always set to the root post. For example, in the post/page/template editors, it will be set with the postId of the loaded post, and the postType as well. We still need that context to be set at a root level in those editors :/

So obviously in the above editors, postId and postType in the post content block will always match the root context and therefore the block will render a placeholder.

In the site editor, though, there is no root post set at all. when you change pages in the nav menu, we just update the root context with that new postId and postType. These values are always different from `undefined`, so it renders the correct thing.

I'm not sure the infinite recursion is related to context, IMO. As far as I can tell, it is only related to a block which could possibly also show up again in the innerblocks of itself, but happens to pull that innerblock list from the same source. At the moment, that's only the post content block.

Do we need more robust checks in `BlockList` to avoid this recursive situation? e.g. an inner block controller attached to a specific entity happens to render the same inner block controller inside of itself, also attached to the same entity.

Reply:
Maybe. Inner blocks and `BlockList` sound like a good lead.

Let's maybe defer the more general solution to later though, and go ahead with this one. (Can we do the [rename](https://github.com/WordPress/gutenberg/pull/24010#discussion_r458068789) though? :pray: :grimacing: )

Reply:
yep, adding that real quick